### ISSUE TITLE
[Question] Why not releasing v3.6.4 with Laravel 7 support?

### ISSUE THREAD

**Issue by hylke94** [jenssegers/laravel-mongodb#2023]
Many projects, including our back-ends use this MongoDB package with Laravel. We would like to update to Laravel 7 with the latest features and other dependencies, but this package holds us back.

Is there any change you guys can release v3.6.4 with the already fixed Laravel 7 support (current develop branch?)? Then we can all upgrade to Laravel 7, while you guys work further on version 4 of this package.

**Comment by divine** [comment]
Hi,

I'm really sorry but we can't simply do that without completing new version.

There is a **huge** problem with **Laravel** itself that we couldn't fix really and I'm still waiting for @Smolevich to check it up.

See https://github.com/laravel/ideas/issues/856

TLDR: there is no point of creating casted class while it doesn't respect on querying a model.

That means  _"Auto-casting ObjectID_" and _"Auto-casting UTCDateTime"_ movement to casted classes that was introduced in 7.0 just vanished.

Taylor just doesn't care (no plans on adding this feature) and I've just gave up on this, so hopefully @Smolevich will have free time and  will check what could be done.

Might be a solution is adding a "fix" to respect casted objects while querying on model.

Thanks.

**Comment by divine** [comment]
Also regarding v3.6.4 this will not happen at all, new version contains a breaking change so next release will be v4 with Laravel 7 support.

**Comment by hylke94** [comment]
But, correct me if I'm wrong, but in all issues i see here, complaining about Laravel 7 support, everyone is saying the current develop branch is working fine. So, why not then?

Now I read it again, you mean the breaking change is the Laravel version support? What about a v4.0.0-alpha version from the current develop branch? That way everyone knows it could break somewhere without support, but we can install it with a fixed version and codebase, without "random" updates on the develop branch. Even the new issue's you could link to this alpha version if it is Laravel v7/pacakge v4 specific.

**Comment by Smolevich** [comment]
I want to need investigate question with auto casting as I can find free time, probably in next 1-2 weeks
@hylke94 We can create alpha release but we can check that it works with laravel 7

**Comment by hylke94** [comment]
@Smolevich I already can't wait for this alpha release ;)

**Comment by datomnurdin** [comment]
Any update for this?

**Comment by divine** [comment]
> Any update for this?

Please, kindly try reading again, might help hopefully.

**Comment by hylke94** [comment]
> > Any update for this?
> 
> Please, kindly try reading again, might help hopefully.

I appreciate that you guys wanna take your time to check it all. But on the other hand, everyone here is reporting in different issues that develop works without any problems.. 

Now, I don't want to push you guys, but an alpha release isn't that big of a deal (I guess, didn't have to do it myself yet). It's an alpha, so theoretically not ready for production, but a risk I will take for our new internal applications. But I would want a tag/version to rely on, instead of the develop commits. 

No pressure tho ;)

**Comment by hipig** [comment]
My current practice is to clone the dev version into my project, which I can only do

**Comment by Smolevich** [comment]
@hylke94 i released alpha version https://github.com/jenssegers/laravel-mongodb/releases/tag/4.0.0-alpha.1
test and write about bugs please
I continue to work on stable version

**Comment by hylke94** [comment]
@Smolevich I will try it first thing tomorrow! Thanks!

**Comment by karpoq** [comment]
Something new about Laravel 7 support? :)

**Comment by hylke94** [comment]
> Something new about Laravel 7 support? :)

The alpha version is compatible with Laravel 7. Have it running with great success here for the last two weeks!
